### DISCUSSION TITLE
🔗 Display only relevant taxonomies according to context or query filter (Related to #23185, #23341, #23361)

### DISCUSSION THREAD

**Discussion by luxmediq** [elementor/elementor#23418] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

I have tried to use the taxonomie filter on a custom product loop on a product catergory page. The issue is that the taxonomie filter will display all taxonomies or categories available on the woocommerce site without taking the context into account. 

Example: I have 3 categories in my shop: Shoes, Shirts, Pants. On my Shoes categories page, the taxonomie filter (e.g., set to categories), will display Shoes, Shirts, Pants.

### Describe the solution you'd like

It would be nice if the taxonomie filter would respect the context of the page (or the query on the custom loop), or have the possibility to define the scope of what values are displayed in the taxonomie filter.

Desired behavior: If custom loop filters on category shoes, then the taxonomies displayed in the taxonomie filter should be scoped to include only relevant taxonomies (e..g subcategories of shoes, taxonomies assigned to products in the categorie of shoes)

### Describe alternatives you've considered

I have not found any alternative with Elementor. Other plugins are the only easy alternative.

**Comment by nicholaszein** [comment]
Hello @luxmediq,

Thanks for contributing your suggestion! 🙏

Your ideas and input as a Web Creator are valuable to us. We're always keen to explore ways to enhance Elementor to meet our community's needs better.

#### 💡 Due to the high volume of feature requests we receive and our commitment to continuous improvement, we can't guarantee that every suggestion will be immediately included in our roadmap.

We appreciate your patience as we take the time to review your request thoroughly. We aim to ensure all decisions align with our mission to provide Web Creators with the tools they need to build their dream websites.

Kind regards

**Reply by JennyPower** [reply]
The absence of this functionality produces 'an incorrect or unexpected result' as the filter doesn't adhere to the query of the connected loop grid - so wouldn't this qualify as a bug?

**Reply by nicholaszein** [reply]
Not really @JennyPower, because it wasn't built to do what you are expecting. The first iteration of the taxonomy filter was built to reproduce exactly how the filters in the Portfolio widget work. Now, with the feedback of everyone we are making plans to expand its functionality.

A bug is an error in the code that prevents something from working as expected based on the code written. The functionality requested here wasn't built into the widget's code.

**Reply by koonda** [reply]
So again you deployed "half-finished" feature. Amazing. With this pace we will get bullet-proof funcitionality by end of 2026.

@nicholaszein why don't you focus on releasing **one** fully working feature instead of releasing tons of unusable half-done features?

**Reply by nicholaszein** [reply]
Hey @koonda, we appreciate your feedback (we really do, I'm actually here because I do care), but your comment goes against our [Code of Conduct](https://elemn.to/gh-code-of-conduct).

The Taxonomy Filter was initially built to behave exactly like the filter in the Portfolio widget. Nothing more, nothing less.

We are now gathering you feedback so we can improve the functionality as we see the necessity rising.

Making accusations and being aggressive is not going to expedite the process of development. It'll just be offensive and unnecessary. 

Please, keep the conversation respectful and constructive. There is no need for this kind of aggressive comment. We are aware of the necessity shared by you and your fellow users about needing this extended functionality. And there is already a plan to improve its functionality based on your feedback.

**Comment by nicholaszein** [comment]
This request has duplicates.

- #23392
- #23637
- #23707

**Comment by nicholaszein** [comment]
Related to:
- #23185
- #23341
- #23360
- #23361
- #23417
- #23614
- #23701
- #23865
- #23917

**Comment by davidamarek** [comment]
+1

**Comment by nick-from-elementor** [comment]
+1

**Comment by mackyangeles** [comment]
+1

**Comment by portrayaloflife** [comment]
+1

**Comment by clmnsm** [comment]
+1

**Comment by maxms** [comment]
+1

**Comment by tommmoe** [comment]
+1 would be a huge improvement to category archive pages

**Comment by design219** [comment]
+1. The taxonomy filter is of limited use at this stage.

**Comment by saawood13** [comment]
Taxonomy filter is not working - my problem is described exactly in this reddit thread - https://www.reddit.com/r/elementor/comments/17qgy5z/no_keywords_are_visible_in_the_taxonomy_filter/

**Comment by yyahav** [comment]
+1

**Comment by dregwide** [comment]
+1

**Comment by heroicP** [comment]
+1

**Comment by Arakaki-code** [comment]
+ 1

**Comment by Vossha** [comment]
+1

**Comment by sophel93** [comment]
+1

**Comment by forrestwentzel** [comment]
+1

**Reply by edcsk** [reply]
+1

**Comment by witty-code** [comment]
+1

**Comment by DanielCyran** [comment]
+1

**Comment by wagnerch1981** [comment]
+1

**Comment by AnselTk1** [comment]
+1

**Comment by javiercriado** [comment]
+1

**Comment by borisdbr** [comment]
+1

**Comment by xdjxdaveyx** [comment]
My solution was to create a custom taxonomy for my posts using the Secures Custom Fields (SCF) plugin and then add this to the posts' (included) taxonomies. The filter picks up the related posts' taxonomies without any problem, and therefore there are no issues with the list being flooded with extra categories or tags from other posts. Any empty taxonomies also get hidden without issue.

**Comment by Rainy93** [comment]
Hi there! I had the same issue, and this is the solution that worked for me: in the query settings (Query > Exclude), I excluded all languages except the one used in the template I’m working on. For instance, if my template is in Spanish, I exclude all other languages. I’m using Polylang, and this fixed the problem for me. I hope it helps you too until Elementor comes up with a full fix. Bye @nicholaszein

**Comment by bluantinoo** [comment]
+1

It's unbelievable a great machine like Elementor lacks this functionality. 

A solid way to filter posts (that should include postmeta along with taxonomies) It's the only missing feature that still forces me to use bloated add-ons on each Woocommerce store I build.

**Comment by MobinDev** [comment]
+1

**Comment by m-vdv** [comment]
+1

**Comment by jpguisa** [comment]
+1

**Comment by stelpy** [comment]
+1, using external Filters in the meantime

**Comment by vishabjeetcoolplugins** [comment]
You might find our free plugin [Loop Grid Extender for Elementor Pro](https://wordpress.org/plugins/loop-grid-extender-for-elementor-pro/) helpful.

It adds include and exclude options for taxonomies inside Elementor’s Loop Grid widget, letting you:
✅ Show only relevant taxonomies (e.g., subcategories of “Shoes” when viewing the Shoes page)
✅ Hide unrelated or empty taxonomies for a cleaner filter experience
✅ Work seamlessly with WooCommerce, posts, or any custom post type

Until Elementor brings this feature natively, this plugin provides an easy, lightweight workaround.

**Comment by tehben** [comment]
+1 Why this isn't part of the roadmap I have no idea. Why it isn't already part of the plugin I have no idea. Why you guys focus on highly technical integrations that only a very small % of users will take advantage of, but leave these basic Woo integrations out, I have no idea.
